src/main/java/ch/uzh/ifi/hase/soprafs21/service/ActivityService.java

```diff
@@ -23,10 +30,40 @@
 
     private final ActivityRepository activityRepository;
 
+    private final ActivityPresetRepository activityPresetRepository;
+
+    private final UserService userService;
+
     @Autowired
-    public ActivityService(@Qualifier("activityRepository") ActivityRepository activityRepository) {
+    public ActivityService(@Qualifier("activityRepository") ActivityRepository activityRepository,
+                           @Qualifier("activityPresetRepository") ActivityPresetRepository activityPresetRepository,
+                           @Qualifier("userService") UserService userService) {
         this.activityRepository = activityRepository;
+        this.activityPresetRepository = activityPresetRepository;
+        this.userService = userService;
+    }
+
+    public List<Activity> getActivities(long userId, String token) {
+        /* ToDo: In the future: check for existing activities that have not been swiped yet (= INITIAL) & add them to returned list */
+        userService.isUserAuthenticated(userId, token);
+        return generateActivities(userId);
     }
 
+    public List<Activity> generateActivities(long userId) {
+        UserSwipeStatus userSwipeStatus = new UserSwipeStatus(userService.getUserByID(userId), SwipeStatus.INITIAL);
+        List<UserSwipeStatus> userSwipeStatusList = Collections.singletonList(userSwipeStatus); // ToDo: This list should contain two users
+
+        List<Activity> generatedActivities = new ArrayList<Activity>();
+        Random rand = new Random();
+
+        /* Generate 10 random activities */
```

Comment:
Shouldn't the generation of activities be based on the categories that the user has chosen instead of generating random activities?